src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/publish.cljs

```diff
@@ -3,55 +3,54 @@
     [org.broadinstitute.firecloud-ui.endpoints :as endpoints]
     ))
 
 
-(defn- render-modal-publish [state refs props]
-  (react/create-element
-    [dialog/OKCancelForm
-     {:header "Publish Method Configuration"
-      :content
-      (react/create-element
-        [:div {}
-         (when (:publishing? @state)
-           [comps/Blocker {:banner "Publishing Method Configuration..."}])
-         (style/create-form-label "Method Configuration Namespace")
-         (style/create-text-field {:style {:width "100%"} :ref "mcNamespace"
-                                   :defaultValue (get-in (:config props) ["namespace"])})
-         (style/create-form-label "Method Configuration Name")
-         (style/create-text-field {:style {:width "100%"} :ref "mcName"
-                                   :defaultValue (get-in (:config props) ["name"])})
-         (when (:error @state)
-           [comps/ErrorViewer {:error (:error @state)}])])
-      :dismiss-self (:dismiss-self props)
-      :ok-button
-      (react/create-element
-        [comps/Button {:text "Publish" :ref "publishButton"
-                       :onClick
-                       #(let [[ns n] (common/get-text refs "mcNamespace" "mcName")]
-                         (swap! state assoc :publishing? true :error nil)
-                         (endpoints/call-ajax-orch
-                           {:endpoint (endpoints/copy-method-config-to-repo (:workspace-id props) (:config props))
-                            :headers {"Content-Type" "application/json"}
-                            :payload  {:configurationNamespace ns,
-                                       :configurationName n,
-                                       :sourceNamespace (get-in (:config props) ["namespace"]),
-                                       :sourceName (get-in (:config props) ["name"])}
-                            :on-done  (fn [{:keys [success? get-parsed-response]}]
-                                        (swap! state dissoc :publishing?)
-                                        (if success?
-                                          ((:dismiss-self props))
-                                          (swap! state assoc :error (get-parsed-response))))}))}])}]))
-
 (react/defc PublishDialog
   {:render
    (fn [{:keys [props state refs]}]
-     [dialog/Dialog
-      {:width 500
-       :dismiss-self (:dismiss-self props)
-       :content (render-modal-publish state refs props)
-       :get-first-element-dom-node #(@refs "mcNamespace")
-       :get-last-element-dom-node #(react/find-dom-node (@refs "publishButton"))}])})
+     (let [{:keys [workspace-id config]} props
+           {:strs [namespace name]} config]
+       [modal/OKCancelForm
+        {:header "Publish Method Configuration"
+         :get-first-element-dom-node #(react/call :access-field (@refs "mcNamespace"))
+         :get-last-element-dom-node #(react/find-dom-node (@refs "publishButton"))
+         :content
```

Comment:
I preferred the previous approach - where the content is its own function - but that's pure style and I won't push for it